src/values.rs

```diff
@@ -660,3 +671,46 @@ fn check_division_by_zero(num: i32) -> Result<()> {
         _ => Ok(()),
     }
 }
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum TransformerValue {
+    Native(fn(Vec<Expression>) -> Vec<Statement>),
+}
+
+impl Display for TransformerValue {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            TransformerValue::Native(_) => write!(f, "<build-in transformer>"),
+        }
+    }
+}
+
+impl TransformerValue {
+    pub fn transform_to_statements(&self, expressions: Vec<Expression>) -> Vec<Statement> {
+        match self {
+            TransformerValue::Native(f) => f(expressions),
+        }
+    }
+    pub fn transform_to_expression(
```

Comment:
This function should placed outside the transformer, or just add an `expect_expression` in `Statement`